lib/src/git.rs

```diff
@@ -1240,6 +1241,7 @@ pub struct GitFetchStats {
 pub fn fetch(
     mut_repo: &mut MutableRepo,
     git_repo: &git2::Repository,
+    depth: Option<NonZeroU32>,
```

Comment:
nit: I would insert `depth` argument somewhere around `callbacks` and `git_settings`. It's not a mandatory option, and we might want to extract a parameter struct later.